src/index.ts

```diff
@@ -0,0 +1,223 @@
+import { IProfile } from "./utils/github/queries/index";
+
+/** @interface Intel defines the structure of the intel class. */
+interface IIntel {
+  /**
+   * Snekclient: A client implementation for snek interaction from the "snek-client" package.
+   */
+  snekclient: SnekClient;
+  /**
+   * @todo Adding error handling.
+   * @function
+   * @param {ISource} source A source object.
+   * @returns {Promise<void>} Empty promise for conformation of completion.
+   * @description Append data to the database based on the information in a source object.
+   */
+  append(source: ISource): Promise<void>;
+  /**
+   * @todo Adding error handling.
+   * @function
+   * @param {ISource[]} sources A list of source objects.
+   * @returns {Promise<void>} Empty promise for conformation of completion.
+   * @description Append data to the databse for n source objects.
+   */
+  appendList?(sources: ISource[]): Promise<void>;
+}
+
+/** @interface Source defines the structure of a source object. */
+interface ISource {
+  /**
+   * User: A username of the provides platforms.
+   */
+  user: string;
+  /**
+   * Platform: A object containing its url and name.
+   */
+  platform: {
+    /**
+     * Optional!
+     * Url: A url of a provided platform.
+     */
+    url?: string;
+    /**
+     * Name: The name of the platform.
+     * It can be chosen according to preferences.
+     */
+    name: string;
+  };
+  /**
+   * Authorization: A token for authorizing the client.
+   */
+  authorization: string;
+}
+
+/** @interface DataUser defines the structure of the reponse of the github client. */
+interface IDataUser {
+  data: {
+    /**
+     * User: Can contain any information according to a user.
+     * The content of this object relays on the query with which the client is addressed.
+     */
+    user: object;
+  };
+}
+//#endregion
+
+//#region > Classes
+/**
+ * @class Intel - A place where everything becomes one thing.
+ * By using the snek-client, utils and the brand new snek-reducer we
+ * created a new way of life.
+ * Some might say that this is not perfect in every way. But they are just fools.
+ * We know every single mistake, every single scratch.
+ * Although we can frankly say, we love it!
+ * @implements IIntel
+ * @see {@link http://github.com/snek-at/intel/README.md |SNEK Intel README} for further information.
+ */
+class Intel implements IIntel {
+  public snekclient: SnekClient;
+  private reducer: Reducer;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @description Creates a instance of the Intel.
+   */
+  constructor() {
+    // init snekclient
+    this.snekclient = new SnekClient();
+    // init reducer
+    this.reducer = new Reducer();
+  }
+
+  /**
+   * Get gitlab or github data and fill the models.
+   *
+   * @param source A source object of type ISource.
+   * @description Fill the models with data of github or gitlab. The type and username is specified by the source param.
```

Comment:
Fill the models with data from github or gitlab. The type and username are specified by the source param.